### PR TITLE
CIAC-8968 - Enhancement Hashicorp Vault integration 

### PR DESCRIPTION
**Pull request by inbalapt1**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.py [hunk 1]
```diff
@@ -99,6 +102,61 @@ def send_request(path, method='get', body=None, params=None, headers=None):
...
+        CommandResults: The command results object containing the retrieved Role ID and role name as outputs.
+    """
+    args = demisto.args()
+    role_name = args.get('role_name')
+    path = f'/auth/approle/role/{role_name}/role-id'
+    response = send_request(path=path, method='get', body={'role_name': role_name})
+    if response:
+        role_id = response.get('data', {}).get('role_id', '')
+    if role_id:
+        return_results(CommandResults(outputs_prefix='HashiCorp.AppRole', outputs={"Id": role_id, "Name": role_name}))
```
Comment (omerKarkKatz) [inline Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.py:157]: What happens if no role_id was found this will not return anything

**Review by omerKarkKatz — COMMENTED** [review]
Hey looks good one comment added please ping me when done, please also add the doc review team as assignee to the PR.

File: Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.yml [hunk 2]
```diff
@@ -338,7 +338,40 @@ script:
...
+    - name: meta_data
+      description: Metadata to be tied to the SecretID. 
+    - name: cidr_list
+      description: Comma separated string or list of CIDR blocks enforcing secret IDs to be used from specific set of IP addresses.
+      isArray: true
+    - name: token_bound_cidrs
+      description: Comma-separated string or list of CIDR blocks.
+      isArray: true
+    - name: num_uses
+      description: Number of times this SecretID can be used, after which the SecretID expires. A value of zero will allow unlimited uses.
```
Comment (richardbluestone) [inline Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.yml:355]: Number of times the SecretID can be used, after which the SecretID expires. A value of zero will allow unlimited uses.

File: Packs/HashiCorp-Vault/Integrations/HashiCorpVault/README.md [hunk 3]
```diff
@@ -404,6 +404,67 @@ Create a new authentication token.
...
+`hashicorp-generate-role-secret`
+#### Input
+
+| **Argument Name** | **Description** | **Required** |
+| --- | --- | --- |
+| role_name | The name of the AppRole. | Required |
+| meta_data | Metadata to be tied to the SecretID. | Optional |
+| cidr_list | Comma separated string or list of CIDR blocks enforcing secret IDs to be used from specific set of IP addresses. | Optional |
+| token_bound_cidrs | Comma-separated string or list of CIDR blocks. | Optional |
+| num_uses | Number of times this SecretID can be used, after which the SecretID expires. A value of zero will allow unlimited uses. | Optional |
```
Comment (richardbluestone) [inline Packs/HashiCorp-Vault/Integrations/HashiCorpVault/README.md:423]: Number of times the SecretID can be used, after which the SecretID expires. A value of zero will allow unlimited uses.

**Review by richardbluestone — APPROVED** [review]

**Pull request merged**